.vscode/settings.json

```
@@ -22,5 +22,6 @@
 		"pojo",
 		"subword"
 	],
-	"files.eol": "\n"
+	"files.eol": "\n",
+	"typescript.tsdk": "node_modules/typescript/lib"
```

This is nice; let's upgrade Typescript in this PR as well though I think

---

I upgraded TS as well, but I left the website at 4.5 as the typedocs only supports up to 4.5 and I didn't want to risk breaking the website!

---

Hmm I think that will be a problem, because we run typedocs on all the code in cursorless to generate api docs. We should prob pin both to the same version

When we switch to nx we'll only have one set of versions for everything fwiw

---

I can try upgrade it but I'm not familiar with docasaurus and it's typedocs so I'd need someone to double check that. It may end up being a much bigger upgrade - if it does need more than a simple dependency upgrade do we want to just do that here or make a different PR for upgrading the website deps.

---

I upgraded TS in website, it seems working to me as far as I can tell. The types are getting output in the contributor docs but I'm not sure they're 100% correct.

---

We should probably also make VSCode actually ask whether we want to use it too, otherwise nobody will actually switch to it.
```suggestion
	"typescript.enablePromptUseWorkspaceTsdk": true,
	"typescript.tsdk": "node_modules/typescript/lib"
```

---

@auscompgeek that sounds reasonable to me.  Any objections @SimeonC ?

Re types in website; I just kicked off a deploy preview.  I'll take a look at docs to see if they look sane

---

Huh.  Unfortunately this PR breaks the api docs.  For example, in the docs for `runCommand`, see how on [production](https://www.cursorless.org/docs/contributing/api/classes/core_commandrunner_commandrunner.commandrunner/#runcommand) there are a bunch of links in the description, but these seem to have disappeared for some reason in the [deploy preview](https://deploy-preview-942--cursorless.netlify.app/docs/contributing/api/classes/core_commandRunner_CommandRunner.CommandRunner/#runcommand) 😕